lib/device.js

```diff
@@ -233,6 +233,27 @@ class Device extends EventEmitter {
 		}
 	}
 
+	async factoryReset() {
```

Comment:
So, I kind of think this might be too targeted of an approach here. Pretty much whenever we want to access something from `particle-usb` directly like in this case we'd need to create a new wrapper method, which seems like too much hassle. How about instead we make an accessor method to get the `_usbDev` instance and just access it directly from tests? + create another method to basically set `this._willDetach` to `true` from the js side of tests as well.

Reply:
Ok, ill give that a shot 👍

Reply:
Updated to expose the USB device directly